scripts/is-release-needed.sh

```diff
@@ -1,13 +1,23 @@
 #!/usr/bin/env bash
 
-# This script checks if the current version of the package is already published on npm
-PACKAGE_VERSION="$(node -p -e "require('./package.json').version")"
-PACKAGE_NAME="$(node -p -e "require('./package.json').name")"
+# This script takes the package directory as an argument
+PACKAGE_DIR=$1
 
-FOUND_VERSION=$(npm view $PACKAGE_NAME versions | grep $PACKAGE_VERSION)
+# Change to the specific package directory
+cd $PACKAGE_DIR
+
+# Check if the current version of the package is already published on npm
+PACKAGE_VERSION=$(node -p "require('./package.json').version")
+PACKAGE_NAME=$(node -p "require('./package.json').name")
+
+echo "Checking if $PACKAGE_NAME@$PACKAGE_VERSION is already published..."
+
+FOUND_VERSION=$(npm view $PACKAGE_NAME versions --json | grep -o "\"$PACKAGE_VERSION\"")
 
 if [ -z "$FOUND_VERSION" ]; then
     IS_NEW_VERSION=true
+else
+    IS_NEW_VERSION=false
 fi
 
-echo "is-release-needed=$(echo $IS_NEW_VERSION)"
+echo "is-release-needed=$IS_NEW_VERSION" >> $GITHUB_OUTPUT
```

Comment:
I don't think this will work. I think the `>> $GITHUB_OUTPUT` needs to be in the workflow yml file, not in this script.

Also, to accommodate the `matrix` I think we should use `exit 0` for success and `exit 1` for failure.